### PR TITLE
Homework: Vector of shared ptrs

### PR DESCRIPTION
**Pull request by MalarzDawid**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: vector-of-shared-ptrs
🏅 Granted 5 XP for @MalarzDawid

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#include "vectorFunctions.hpp"
+#include <vector>
+#include <memory>
+#include <iostream>
+
+std::vector<std::shared_ptr<int>> generate(int count) {
+    std::vector<std::shared_ptr<int>> vec;
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:7]: Reserve memory for vec

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    for (size_t i = 0; i < count; i++ ) {
+        vec.push_back(std::make_shared<int>(i));
+    }
+
+    return vec;
+}
+
+void print(std::vector<std::shared_ptr<int>> vec) {
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:16]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    for (auto el: vec) {
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:17]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        std::cout << el << "\n";
+    }
+}
+
+void add10(std::vector<std::shared_ptr<int>> vec) {
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:22]: Pass by const refrence. Try always avoid copy.

File: homework/vector-of-shared-ptrs/vectorFunctions.cpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    for (size_t i = 0; i < count; i++ ) {
+        vec.push_back(std::make_shared<int>(i));
+    }
+
+    return vec;
+}
+
+void print(std::vector<std::shared_ptr<int>> vec) {
+    for (auto el: vec) {
+        std::cout << el << "\n";
```
Comment (nauka-programowania-MA) [inline homework/vector-of-shared-ptrs/vectorFunctions.cpp:18]: Here you print shared_ptr not element pointed by them

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**